main.go

```diff
@@ -595,17 +608,20 @@ func mostLikedColorChanged() bool {
 	if opinions[multiverse.Blue] > opinions[multiverse.Green] {
 		currentMostLikedColor = multiverse.Blue
 	}
-	if opinions[multiverse.Red] > opinions[multiverse.Blue] {
+	if opinions[multiverse.Red] > opinions[multiverse.Blue] && opinions[multiverse.Red] > opinions[multiverse.Green] {
 		currentMostLikedColor = multiverse.Red
 	}
-	if mostLikedColor != currentMostLikedColor && opinions[mostLikedColor] > opinions[currentMostLikedColor] {
-		// color set for the first time
+	// color selected
+	if mostLikedColor != currentMostLikedColor {
+		// color selected for the first time, it not counts
 		if mostLikedColor == multiverse.UndefinedColor {
 			mostLikedColor = currentMostLikedColor
 			return false
 		}
-		mostLikedColor = currentMostLikedColor
-		return true
+		if opinions[currentMostLikedColor] > opinions[mostLikedColor] {
```

Comment:
Is this redundant? in Lines 605- 613 we have already gotten 
- `currentMostLikedColor` has most opinions
, and in Line 615 we have the gotten
- `currentMostLikedColor` != `mostLikedColor`
From the above two, does not it imply `opinions[currentMostLikedColor] > opinions[mostLikedColor]`?